modules/desktop/profiles/applications.nix

```diff
@@ -4,21 +4,112 @@
 {
   config,
   lib,
+  pkgs,
   ...
 }: let
   cfg = config.ghaf.profiles.applications;
-in {
-  options.ghaf.profiles.applications = {
-    enable = lib.mkEnableOption "Some sample applications";
-    #TODO Create options to allow enabling individual apps
-  };
+in
+  with lib; {
+    options.ghaf.profiles.applications = {
+      enable = mkEnableOption "Some sample applications";
+      #TODO Create options to allow enabling individual apps
+      #weston.ini.nix mods needed
+      ivShMemServer = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = "Enable memory sharing between virtual machines";
+        };
+        memSize = mkOption {
+          type = lib.types.int;
+          default = 16;
+          description = mdDoc ''
+            Defines shared memory size
+          '';
+        };
+        vmCount = mkOption {
+          type = lib.types.int;
+          default = 6;
+          description = mdDoc ''
+            Defines maximum number of application VMs
+          '';
+        };
+        serverSocketPath = mkOption {
+          type = lib.types.str;
+          default = "/run/user/1000/memsocket-server.sock";
+          description = mdDoc ''
+            Defines location of the listening socket.
+            It's used by waypipe as an output socket when running in server mode
+          '';
+        };
+        clientSocketPath = mkOption {
+          type = lib.types.str;
+          default = "/run/user/1000/memsocket-client.sock";
+          description = mdDoc ''
+            Defines location of the output socket. It's fed
+            with data coming from AppVMs.
+            It's used by waypipe as an input socket when running in client mode
+          '';
+        };
+        hostSocketPath = mkOption {
+          type = lib.types.str;
+          default = "/tmp/ivshmem_socket"; # The value is hardcoded in the application
+          description = mdDoc ''
+            Defines location of the shared memory socket. It's used by qemu
+            instances for memory sharing and sending interrupts.
+          '';
+        };
+        flataddr = mkOption {
+          type = lib.types.str;
+          default = "0x920000000";
+          description = mdDoc ''
+            If set to a non-zero value, it maps the shared memory
+            into this physical address.
```

Comment:
Would you please add to `description` on how the default value was/is chosen?

Reply:
Done.